lib/linguist/heuristics.rb

```diff
@@ -274,7 +274,7 @@ def call(data)
     end
 
     disambiguate ".pl" do |data|
-      if /^(use v6|(my )?class|module)/.match(data)
+      if /^(use v6|(my)? class|module)/.match(data) #class might be a variable name in Perl 5
```

Comment:
> I'm not sure I follow you... What do you mean by that?

First, Any.pm which I have added in the test suite indeed is recognized as Perl6, which is it not. Since it does not have module or v6, it must be because line 49 fires that regular expression. Please note that ^ might not be working as intended, since *anyway* `class` or `module` or `use v6` might not be at the very beginning of a line, but indented by whitespace. So my hunch is that the beginning of a line ^ is anchored to the optional `my`. If that is not pressent, it fires whenever it finds `class` anywhere in a line (which indeed it does). 
Anyway, experience is the mother of science. Please check the file I provide (`Any.pm`) with the former and current regex. If it checks out as perl6, then it's wrong. If it's perl5, then it's correct! :-)